src/solve.jl

```diff
@@ -210,10 +210,13 @@ function DiffEqBase.__init(
   end
   rateType = typeof(rate_prototype) ## Can be different if united
 
-  if prob.f isa DynamicalSDEFunction
-    noise_rate_prototype = rate_prototype.x[1]
-  elseif is_diagonal_noise(prob)
-    noise_rate_prototype = rate_prototype
+
+  if is_diagonal_noise(prob)
+    if prob.f isa DynamicalSDEFunction
+      noise_rate_prototype = rate_prototype.x[1]
```

Comment:
That seems like an odd way to put it. In other words, it's diagonal on one part of the system?

Reply:
I wonder if we just shouldn't classify that as diagonal noise at all, but it looks like it doesn't lead to too many required changes.

Reply:
But the proposed change here is to allow DynamicalSDEFunction to have non diagonal noise, as the previous implementation assume the noise was diagonal.